otter/convergence/steps.py

```diff
@@ -50,6 +50,69 @@ def set_server_name(server_config_args, name_suffix):
     return server_config_args.set_in(('server', 'name'), name)
 
 
+def _try_json_message(maybe_json_error, keys):
```

Comment:
Will do - do you want me to try to make it slightly more generic?  Like `UpstreamError` just grabs the first outer key, so maybe we can add some utility to handle that instead of just a specific key like "badRequest" ?